WordPress/openverse-catalog
# Refactor Science museum to use ProviderDataIngester

## Fixes


Fixes #582

## Description


This was PR created to better review the new Ingester class and to show how awesome it is :)
Now it's the actual PR to refactor the Science museum script to use ProviderDataIngester class. It also should fix the issue when the license information was incorrect and caused the script to stop.

While refactoring this script, I thought that we are probably extracting the creator here incorrectly:

Science museum uses the "Maker" of the object as the author for the image. Shouldn't we normally put the **author** of the **image**, not the **object** being photographed? 

For example, for this telescope replica(https://collection.sciencemuseumgroup.org.uk/objects/co56202/telescope-by-galileo-replica-telescope-galilean-telescope-refracting-replica) should we put _Galileo_ or  _The Board of Trustees of the Science Museum_  (based on the note below the image: Science Museum Group Collection © The Board of Trustees of the Science Museum )? 

## Testing Instructions

`just test` and confirm tests pass.

Run the Science Museum provider script and watch the pull_data task. Make sure it at least pulls records for a few buildings before marking as success; alternatively you can set an ingestion_limit admin variable to cap the number of records pulled.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
